src/lib/operators/join_sort_merge.cpp

```diff
@@ -355,13 +385,110 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
     }
 
     // Join the rest of the unfinished side, which is relevant for outer joins and non-equi joins
-    auto right_rest = TableRange(cluster_number, right_run_start, right_size);
-    auto left_rest = TableRange(cluster_number, left_run_start, left_size);
+    const auto right_rest = TableRange(cluster_number, right_run_start, right_size);
+    const auto left_rest = TableRange(cluster_number, left_run_start, left_size);
     if (left_run_start < left_size) {
       _join_runs(left_rest, right_rest, CompareResult::Less);
     } else if (right_run_start < right_size) {
       _join_runs(left_rest, right_rest, CompareResult::Greater);
     }
+
+    // Short cut implementation for Anti joins. Implementing anti joins within the current sort-merge join
```

Comment:
How can this be a short cut if the method ends right below/

Reply:
Short cut in a sense that we did not implement the anti join within the "actual" sort-merge join, but as a postprocessing step where we negate the semi-join result. Happy to take better terms.